hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -243,9 +243,37 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 
 	// Get HostedCluster deps.
 	controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
-	ignEndpoint := hcluster.Status.IgnitionEndpoint
 	infraID := hcluster.Spec.InfraID
 
+	// Loop over all conditions.
+	// Order matter as conditions might choose to short circuit returning ctrl.Result or error.
+	signalConditions := map[string]func(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster, conditionType string) (*ctrl.Result, error){
```

Comment:
why do we need to pass conditionType to the functions, shouldn't each function know which condition it is updating?

Reply:
yes, it's not programmatically needed. It seemed easier to grasp at a glance to me what conditions are being applied. They can also be used to unit test everything in the index end up set.
I'm happy to change the approach to something else if you prefer.